javaparser-core/src/main/java/com/github/javaparser/utils/CodeGenerationUtils.java

```diff
@@ -112,17 +113,28 @@ public static String packageToPath(String pkg) {
 
     /**
      * Calculates the path of a package.
+     * <p><b>Note:</b> The {@link Path} is always using the {@link FileSystems#getDefault() default filesystem}.
      *
      * @param root the root directory in which the package resides
-     * @param pkg the package, like "com.laamella.parser"
+     * @param pkg  the package, like "com.laamella.parser"
+     * @return the {@link Path} of the package's directory
      */
     public static Path packageAbsolutePath(String root, String pkg) {
         pkg = packageToPath(pkg);
         return Paths.get(root, pkg).normalize();
     }
 
+    /**
+     * Calculates the path of a package.
+     * <p><b>Note:</b> Unlike {@link #packageAbsolutePath(String, String)}, the {@link Path} will use the same filesystem as the given path.
+     *
+     * @param root the root directory in which the package resides
+     * @param pkg  the package, like "com.laamella.parser"
+     * @return the {@link Path} of the package's directory
+     */
     public static Path packageAbsolutePath(Path root, String pkg) {
-        return packageAbsolutePath(root.toString(), pkg);
+        pkg = packageToPath(pkg);
```

Comment:
Is there a change in behavior compared to the previous version of the method?

Reply:
No, as per the docs of [`Paths.get`](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html#get(java.lang.String,%20java.lang.String...)):
>  Note that while this method is very convenient, using it will imply an assumed reference to the default FileSystem and limit the utility of the calling code. Hence it should not be used in library code intended for flexible reuse. A more flexible alternative is to use an existing Path instance as an anchor, such as:

     Path dir = ...
     Path path = dir.resolve("file");
 
Replacing `Paths.get` by `Path#resolve` is the same, except the `FileSystem` being used is preserved

Reply:
I will leave this open as a draft for the moment, my code isn't giving the same results when using the ZIP and I need to figure out why

Reply:
Fixed with the next commits